bb-junit5/src/main/java/com/cognifide/qa/bb/junit5/guice/GuiceExtension.java

```diff
@@ -0,0 +1,198 @@
+/*-
+ * #%L
+ * Bobcat
+ * %%
+ * Copyright (C) 2018 Cognifide Ltd.
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.cognifide.qa.bb.junit5.guice;
+
+import static java.util.stream.Collectors.toSet;
+import static org.junit.platform.commons.support.AnnotationSupport.findAnnotation;
+
+import com.google.common.collect.Sets;
+import com.google.inject.AbstractModule;
+import com.google.inject.Guice;
+import com.google.inject.Injector;
+import com.google.inject.Module;
+import java.lang.reflect.AnnotatedElement;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.jupiter.api.extension.ExtensionContext.Store;
+import org.junit.jupiter.api.extension.TestInstancePostProcessor;
+
+/**
+ * Extension that will start guice and is responsible for the injections to test instance Based on
+ * <a href="https://github.com/JeffreyFalgout/junit5-extensions/tree/master/guice-extension">Guice
+ * Extension</a>
+ */
+public class GuiceExtension implements TestInstancePostProcessor {
+
+  private static final Namespace NAMESPACE =
+      Namespace.create("com", "cognifide", "qa", "bb", "junit", "guice");
+
+  @Override
+  public void postProcessTestInstance(Object testInstance, ExtensionContext context)
+      throws Exception {
+
+    getOrCreateInjector(context).ifPresent(injector -> injector.injectMembers(testInstance));
+
+  }
+
+  /**
+   * Create {@link Injector} or get existing one from test context
+   */
+  private static Optional<Injector> getOrCreateInjector(ExtensionContext context)
+      throws NoSuchMethodException, InstantiationException, IllegalAccessException, InvocationTargetException {
+
+    Optional<AnnotatedElement> optionalAnnotatedElement = context.getElement();
+    if (!optionalAnnotatedElement.isPresent()) {
+      return Optional.empty();
+    }
+
+    AnnotatedElement element = optionalAnnotatedElement.get();
+    Store store = context.getStore(NAMESPACE);
+
+    Injector injector = store.get(element, Injector.class);
```

Comment:
There is a `#getOrComputeIfAbsent` method available here, see the comment below.